src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3Connection.cs

```diff
@@ -295,15 +295,22 @@ private void OnMaximumStreamCountIncrease(long newMaximumStreamCount)
                     return;
                 }
 
-                _requestStreamsRemaining += (newMaximumStreamCount - _maximumRequestStreams);
+                IncreaseRemainingStreamCount(newMaximumStreamCount - _maximumRequestStreams);
                 _maximumRequestStreams = newMaximumStreamCount;
+            }
+        }
+
+        private void IncreaseRemainingStreamCount(long delta)
```

Comment:
if `delta` is negative "AdjustRemainingStreamCount" may be better name.

Reply:
It's never `<= 0`. I added an assert to express that.